### ISSUE TITLE
"Metadata allocator corruption: allocation is NULL" with Swift 5.6.2 on focal arm64

### ISSUE THREAD

**Issue by dnadoba** [apple/swift#60169]
**Describe the bug**
Note: **This seems to be already fixed in `swiftlang/swift:nightly-5.7-focal` and `swiftlang/swift:nightly-main-focal`**. 

When I run `swift-nio`s test with `swift test --parallel` in docker with `swift:5.6.2-focal` on an M1 Mac, test execution crashes with the following output:
```
[1449/1449] Testing NIOConcurrencyHelpersTests.NIOConcurrencyHelpersTests/testLargeContendedNIOAtomicSum

Test Suite 'Selected tests' started at 2022-07-21 10:55:53.146
Test Suite 'ChannelTests' started at 2022-07-21 10:55:53.147
Test Case 'ChannelTests.testAcceptsAfterCloseDontCauseIssues' started at 2022-07-21 10:55:53.147
Metadata allocator corruption: allocation is NULL. curState: {(nil), 16088} - curStateReRead: {(nil), 16088} - newState: {0xa8, 15920} - allocatedNewPage: false - requested size: 168 - sizeWithHeader: 168 - alignment: 8 - Tag: 13

Exited with signal code 6
```

I think that we hit the fatalError here: https://github.com/apple/swift/blob/8ca0f47e770c8351f21f88d17c261830692f6caf/stdlib/public/runtime/Metadata.cpp#L6338-L6350

I can only reproduce this with Swift 5.6.2 in docker. Seems to be already fixed in 5.7+.

**Steps To Reproduce**
```
git clone https://github.com/apple/swift-nio
cd swift-nio
docker run --rm --volume "$(pwd)/:/src" --workdir "/src/" --platform=linux/arm64 swift:5.6.2-focal swift test --parallel 
```

**Expected behavior**
Tests should not crash and pass

**Environment (please fill out the following information)**
 - OS: macOS 13.0 Beta (22A5295h) but in docker 
 - Swift: `swift:5.6.2-focal`

**Additional context**
Doesn't appear to be happening if `--parallel` is not specified.
Doesn't happen on newer swift versions. 
Might be the same issue hit by https://github.com/apple/swift/issues/58308 and potentially already fixed in https://github.com/apple/llvm-project/pull/3917 but not sure if this fix is already included in Swift 5.6.2 or only in Swift 5.7.

**Comment by Luffy19960106** [comment]
are there some action to avoid this crash？I can‘t suggest user to update system when meet this crash in iOS 15.4，and threre are a lot of iOS15.4 users

**Comment by AnthonyLatsis** [comment]
I cannot tell which version of the Swift runtime shipped with iOS 15.4.1, or which macOS version got the fixed Swift runtime, but the issue was said to have been fixed in 5.7 and it’s been a while, so I’m going to close this as a duplicate of #58308.
